src/System.Device.Gpio/System/Device/Gpio/Drivers/SunxiDriver.Linux.cs

```diff
@@ -0,0 +1,470 @@
+        {
+            GpioRegisterOffset0 = gpioRegisterOffset0;
+            GpioRegisterOffset1 = gpioRegisterOffset1;
+        }
+
+        /// <summary>
+        /// The number of pins provided by the driver.
+        /// </summary>
+        protected internal override int PinCount => throw new PlatformNotSupportedException("This driver is generic so it can not enumerate how many pins are available.");
+
+        /// <summary>
+        /// Converts a board pin number to the driver's logical numbering scheme.
+        /// </summary>
+        /// <param name="pinNumber">The board pin number to convert.</param>
+        /// <returns>The pin number in the driver's logical numbering scheme.</returns>
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        /// <summary>
+        /// Opens a pin in order for it to be ready to use.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        protected internal override void OpenPin(int pinNumber)
+        {
+            Initialize();
+            SetPinMode(pinNumber, PinMode.Input);
+        }
+
+        /// <summary>
+        /// Closes an open pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        protected internal override void ClosePin(int pinNumber)
+        {
+            if (_sysFSModes.ContainsKey(pinNumber) && _sysFSModes[pinNumber] == PinMode.Output)
+            {
+                Write(pinNumber, PinValue.Low);
+                SetPinMode(pinNumber, PinMode.Input);
+            }
+        }
+
+        /// <summary>
+        /// Sets the mode to a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (!IsPinModeSupported(pinNumber, mode))
+            {
+                throw new InvalidOperationException($"The pin {pinNumber} does not support the selected mode {mode}.");
+            }
+
+            // Get port controller, port number and shift
+            var unmapped = UnmapPinNumber(pinNumber);
+            int cfgNum = unmapped.Port / 8;
+            int cfgShift = unmapped.Port % 8;
+            int pulNum = unmapped.Port / 16;
+            int pulShift = unmapped.Port % 16;
+
+            // Get register address, register pointer
+            int cfgAddress, pulAddress;
+            uint* cfgPointer, pulPointer;
+            if (unmapped.PortController < 10)
+            {
+                cfgAddress = (GpioRegisterOffset0 + unmapped.PortController * 0x24 + cfgNum * 0x04) & _mapMask;
```

Comment:
nit: might be worth to move `unmapped.PortController * 0x24 + cfgNum * 0x04) & _mapMask` and expression below to a local